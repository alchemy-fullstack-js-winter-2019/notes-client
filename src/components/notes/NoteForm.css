.NotesForm {
  margin: 1em auto;
  width: 50%;
  max-width: 500px;

  input {
    display: block;
    align-self: center;
    width: 50%;
    max-width: 250px;
    margin: auto;
    padding: 0.4em;
  }

  textarea {
    display: block;
    width: 100%;
    max-width: 500px;
    height: 100vh;
    max-height: 250px;
    margin: 0.5em;
    padding: 0.4em;
  }

  button {
    display: block;
    width: 100%;
    margin: 0.5em;
    padding: 0.4em;
    border: 0;

    background-color: #2D1050;
    color: #E9E2E4;

    font-weight: 700;

    cursor: pointer;
  }

  button:disabled {
    background-color: rgba(80,29,144, 0.4);
    cursor: wait;
  }

  .error {
    padding: 0.4em 1em;
    margin: 1em;
    border-radius: 0.4em;

    background-color: #A91310;
    color: #E9E2E4;
  }
}
