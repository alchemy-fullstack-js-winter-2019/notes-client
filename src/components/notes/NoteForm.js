import React from 'react';
import PropTypes from 'prop-types';
import { FaAsterisk } from 'react-icons/fa';
import styles from './NoteForm.css';

function NotesForm({ title, body, error, loading, createNote, titleChange, bodyChange }) {
  return (
    <form className={styles.NotesForm} onSubmit={createNote.bind(null, title, body)}>
      {error && <section className={styles.error}>
        <p>Error: {error}</p>
      </section>}
      <input
        type="text"
        name="title"
        placeholder="Note Title"
        value={title}
        onChange={titleChange}
      />
      <textarea
        name="body" value={body}
        onChange={bodyChange}
      ></textarea>
      <button disabled={loading}>
        <span className={loading ? styles.spinner : ''}>{loading ? <FaAsterisk /> : 'Create'}</span>
      </button>
    </form>
  );
}

NotesForm.propTypes = {
  error: PropTypes.string,
  loading: PropTypes.bool.isRequired,
  title: PropTypes.string.isRequired,
  body: PropTypes.string.isRequired,
  createNote: PropTypes.func.isRequired,
  titleChange: PropTypes.func.isRequired,
  bodyChange: PropTypes.func.isRequired
};

export default NotesForm;
