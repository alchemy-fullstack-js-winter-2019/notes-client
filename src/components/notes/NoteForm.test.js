import React from 'react';
import { shallow } from 'enzyme';
import NoteForm from './NoteForm';

describe('Note tests', () => {
  let wrapper;
  let createNote;
  beforeEach(() => {
    createNote = jest.fn();
    wrapper = shallow(
      <NoteForm
        title="My Title"
        body="A body"
        error={null}
        loading={false}
        createNote={createNote}
        titleChange={jest.fn()}
        bodyChange={jest.fn()}
      />
    );
  });
  it('matches a snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('calls the createNote function on sumbit', () => {
    wrapper.find('form').simulate('submit', {
      preventDefault: jest.fn()
    });

    expect(createNote).toHaveBeenCalled();
  });
});
