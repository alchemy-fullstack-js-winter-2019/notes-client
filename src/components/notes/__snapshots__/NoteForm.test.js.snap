// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note tests matches a snapshot 1`] = `undefined`;
